### ISSUE TITLE
It's a route param - not query string. If the parameter doesn't exist it means the route is different.

### ISSUE THREAD

**Issue by luke-hanwook** [luke-hanwook/github-issue-management-practice#33]
It's a route param - not query string. If the parameter doesn't exist it means the route is different.
Let's say you have this configuration: `/group/:groupId/post/:postId`
If groupId is undefined the route will be `/group/post/<postId>` and if postId is undefined the route will be `/group/<groupId>/post`
These two examples should match different routes than `/group/:groupId/post/:postId`

A common convention is that path/route params are manditory while query params are optional.

_Originally posted by @tzachov in https://github.com/remix-run/react-router/issues/8498#issuecomment-997176141_

**Comment by micmcg** [comment]
Yep I'm hitting this issue also. The fact that the route has matched should mean that a param is **always** defined.

**Comment by yeliex** [comment]
the component should decide it can receive optionally params actually

eg:
```
const Page = () => {
    const params = useParams<{ id1: string; id2?: string }>;
}

const route = [
  {
      path: '/item/:id1/item2/:id2',
      Component: Page
  }, 
  {
    path: '/item/:id1',
    Component: Page
  }
];
```

the case should be defined in route manually and could be known by developer
